MidasFrontend/ArduinoHelper.cxx

```diff
@@ -0,0 +1,195 @@
+#include "ArduinoHelper.h"
+#include "TestStandMessages.h"
+#include "Gantry.h"
+
+#include "LinuxSerialDevice.h"
+#include "TestStandCommHost.h"
+
+#include "shared_defs.h"
+
+#include "midas.h"
+
+#include <stdio.h>
+#include <math.h>
+
+#define BAUD_RATE           115200
+#define MSG_RECEIVE_TIMEOUT 5000
+
+LinuxSerialDevice device;
+TestStandCommHost comm(device);
+
+// Gantry Mechanical Properties
+const float pulley_diameter_mm   = 17;
+const float dist_per_rev_mm      = 2 * M_PI * (pulley_diameter_mm / 2);
+const float mm_cts_ratio         = dist_per_rev_mm / ENCODER_COUNTS_PER_REV;
+const float mm_steps_ratio       = dist_per_rev_mm / MOTOR_STEPS_PER_REV;
+
+// Operational Bounds
+const float gantry_x_min_mm      = 0.0;
+const float gantry_x_max_mm      = 1200.0; // max rail is 1219 mm
+const float gantry_y_min_mm      = 0.0;
+const float gantry_y_max_mm      = 1200.0;
+const float gantry_vel_min_mm_s  = 0.0; 
+const float gantry_vel_max_mm_s  = 10.0;
+
+/*****************************************************************************/
+/*                             PRIVATE FUNCTIONS                             */
+/*****************************************************************************/
+
+static bool validate_move_params(float *dest_mm, float *vel_mm_s)
+{
+    if (dest_mm[AXIS_X] < gantry_x_min_mm || dest_mm[AXIS_X] > gantry_x_max_mm) {
+        cm_msg(
+            MERROR,
+            "validate_move_params",
+            "Destination on x-axis should be between %f mm and %f mm inclusive.",
+            gantry_x_min_mm, gantry_x_max_mm);
+        return false;
+    }
+
+    if (dest_mm[AXIS_Y] < gantry_y_min_mm || dest_mm[AXIS_Y] > gantry_y_max_mm) {
+        cm_msg(
+            MERROR,
+            "validate_move_params",
+            "Destination on y-axis should be between %f mm and %f mm inclusive.",
+            gantry_y_min_mm, gantry_y_max_mm);
+        return false;
+    }
+
+    if (vel_mm_s[AXIS_X] < gantry_vel_min_mm_s || vel_mm_s[AXIS_X] > gantry_vel_max_mm_s
+        || vel_mm_s[AXIS_Y] < gantry_vel_min_mm_s || vel_mm_s[AXIS_Y] > gantry_vel_max_mm_s) {
+        cm_msg(
+            MERROR,
+            "validate_move_params",
+            "Velocity should be between %f mm/s and %f mm/s inclusive.",
+            gantry_vel_min_mm_s, gantry_vel_max_mm_s);
+        return false;
+    }
+
+    return true;
+}
+
+static int32_t mm_to_cts(float val_mm) {
+    return round(val_mm / mm_cts_ratio);
+}
+
+static uint32_t mm_to_steps(float val_mm) {
+    return round(val_mm / mm_steps_ratio);
+}
+
+static AxisDirection get_direction(int32_t displacement) {
+    return (displacement < 0 ? AXIS_DIR_NEGATIVE : AXIS_DIR_POSITIVE);
```

Comment:
had to look up what `?` did, pretty neat!

Reply:
Will abuse it now HAHAHA

Reply:
Oh no, what have I done!